test_runner_lib/include/test_runner/TestRunner.h

```diff
@@ -114,13 +116,23 @@ void ExecuteLiftedFunction(llvm::Function *func, size_t insn_length, T *state,
           target->getName().str());
 
   assert(returned != nullptr);
-  auto orig_pc = state->gpr.rip.dword;
+  auto orig_pc = program_counter_fetch(state);
   // run until we terminate and exit pc
-  while (state->gpr.rip.dword == orig_pc) {
-    returned(state, state->gpr.rip.dword, handler);
+  while (program_counter_fetch(state) == orig_pc) {
+    returned(state, program_counter_fetch(state), handler);
   }
 }
 
+template <typename T>
+void RandomizeState(T *state, random_bytes_engine &rbe) {
```

Comment:
Why is `state` a pointer? Does `nullptr` make sense here? If not -> use ref instead.